x-pack/plugins/observability_solution/observability_ai_assistant/server/rule_connector/index.ts

```diff
@@ -0,0 +1,248 @@
+    client,
+    screenContexts: [],
+  });
+  const actionsClient = await (
+    await resources.plugins.actions.start()
+  ).getActionsClientWithRequest(request);
+
+  const systemMessage = buildSystemMessage(functionClient, execOptions.params.rule, state);
+
+  await lastValueFrom(
+    client
+      .complete({
+        functionClient,
+        persist: true,
+        isPublic: true,
+        connectorId: execOptions.params.connector,
+        signal: new AbortController().signal,
+        kibanaPublicUrl: (await resources.context.core).coreStart.http.basePath.publicBaseUrl,
+        messages: [
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.System,
+              content: systemMessage,
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.User,
+              content: execOptions.params.message,
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.Assistant,
+              content: '',
+              function_call: {
+                name: 'get_connectors',
+                arguments: JSON.stringify({}),
+                trigger: MessageRole.Assistant as const,
+              },
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.User,
+              name: 'get_connectors',
+              content: JSON.stringify({
+                connectors: await actionsClient.getAll().then((connectors) => {
+                  return connectors.map((connector) => {
+                    if (connector.actionTypeId === '.slack') {
+                      return {
+                        ...connector,
+                        // TODO: figure out how to parse ParamsSchema at
+                        // x-pack/plugins/stack_connectors/server/connector_types/slack/index.ts
+                        params: {
+                          message: 'string',
+                        },
+                      };
+                    }
+
+                    return connector;
+                  });
+                }),
+              }),
+            },
+          },
+        ],
+      })
+      .pipe(
+        filter(
+          (event): event is ChatCompletionChunkEvent =>
+            event.type === StreamingChatResponseEventType.ChatCompletionChunk
+        )
+      )
+      .pipe(concatenateChatCompletionChunks())
+  );
+
+  return { actionId: execOptions.actionId, status: 'ok' };
+}
+
+export const getObsAIAssistantConnectorAdapter = (): ConnectorAdapter<
+  ActionParamsType,
+  ConnectorParamsType
+> => {
+  return {
+    connectorTypeId: OBSERVABILITY_AI_ASSISTANT_CONNECTOR_ID,
+    ruleActionParamsSchema: ParamsSchema,
+    buildActionParams: ({ params, rule, ruleUrl, alerts }) => {
```

Comment:
My last-minute PR (https://github.com/elastic/kibana/pull/180437) 🙂 added support to authorize your system actions when creating or editing a rule. You can set the required permissions in your adapter as `getKibanaPrivileges: (params) => ['api:observabilityAIAssistant']`.

Reply:
Added in https://github.com/elastic/kibana/pull/179980/commits/acc5da8b5433511e238238185ffdf6debbafa390